### DISCUSSION TITLE
Automated tests

### DISCUSSION THREAD

**Discussion by Killbrum** [ihhub/fheroes2#6800] [Ideas]
Hello folks! I came up with the question regarding test activities. Have you thought about implementing unit/integration/system/automation tests? I see that the code base has grown, and now since the product has become more mature, there is a high chance of breaking any in-game logic. I've been watching and slightly testing this product for years (well.. yes... I usually stay behind the scene, but I always track every single update). Two years ago the product had a lot of bugs, tons of them (well... you know...) and now it brings an excellent experience. Personally, I have vast engineering experience (developing and managing), and I know how tough it can be to bring a new feature or fix without breaking any previous logic.
So... My question is quite simple.. do you have any plans to bring unit/integration/system/automation tests? I bet it's quite a nice time investment which will save a lot of time in the future.

Thanks for your time!

**Comment by oleg-derevenetz** [comment]
Hi @Killbrum if you have good experience in deploying automated tests and you can implement the appropriate infrastructure for this project, you are welcome to offer a pull request :)
